### PR TITLE
Homework - Modern C++

### PR DESCRIPTION
**Pull request by MaciejSalamonski**

### CODE REVIEW

**Review by WBigaj — APPROVED** [review]
I have checked your code and it looks good :)

**Review by ziobron — CHANGES_REQUESTED** [review]
Good, but delegating constructors are missing. This feature allows you to call constructor of the same class on constructor's initializer list.

**Comment by MaciejSalamonski** [comment]
Delegating constructors added :)

File: shapes/Square.cpp [hunk 1]
```diff
@@ -1,8 +1,11 @@
 #include "Square.hpp"
 #include <iostream>
 
-Square::Square(Color c)
-    : Rectangle(c) {}
+Square::Square(Color color)
+    : Rectangle(color) {}
+
+Square::Square(double x)
+    : Rectangle(x, x, Color::Green) {}
```
Comment (ziobron) [inline shapes/Square.cpp]: ok, but this one is not a delegation. You simply call a base class constructors here. Constructor delegation takes place when you call other constructor of your own class.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
15 XP za Nowoczesny C++ SFINAE (4447)
5 XP za Nowoczesny C++ alignas (4448)
5 XP za Nowoczesny C++ delegating constructors (4449)
5 XP za Nowoczesny C++ lambda (4450)
5 XP za Nowoczesny C++ structured bindings (4451)
15 XP za Nowoczesny C++ variadic templates (4452)

{ "requirement_ids": [4447, 4448, 4449, 4450, 4451, 4452] }

**Pull request closed without merging**